test/plugin/test_out_elasticsearch_dynamic.rb

```diff
@@ -10,8 +11,8 @@ def setup
     @driver = nil
   end
 
-  def driver(tag='test', conf='')
-    @driver ||= Fluent::Test::BufferedOutputTestDriver.new(Fluent::ElasticsearchOutputDynamic, tag).configure(conf)
+  def driver(conf='')
+    @driver ||= Fluent::Test::Driver::Output.new(Fluent::ElasticsearchOutputDynamic).configure(conf)
```

Comment:
It seems that `Fluent::ElasticsearchOutputDynamic` needs to be placed under `Fluent::Plugin` namespace.